nixos/modules/services/web-apps/rimgo.nix

```diff
@@ -0,0 +1,94 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}: let
+  cfg = config.services.rimgo;
+  inherit (lib)
+    mkOption
+    mkEnableOption
+    mkPackageOption
+    mkDefault
+    mkIf
+    types
+    literalExpression
+  ;
+in {
+  options.services.rimgo = {
+    enable = mkEnableOption "rimgo";
+    package = mkPackageOption pkgs "rimgo" {};
+    settings = mkOption {
+      type = with types; attrsOf str;
```

Comment:
```suggestion
      type = with types; attrsOf (nullOr (coercedTo port toString str));
```

Reply:
I have been seeing port as a separate option in a lot of other modules, so I just copied what I saw in other places. 
I also think, this would make it much easier for the port to be invalid accidentally.
Furthermore, if anytime in the future something like RFC 151 (port allocator, so no module tries to use the same port) gets implemented, already having port as a separate option makes it much easier to implement i think.

Reply:
It should be possible to use the `submodule` type to make it possible to have free-form settings with a type-checked subset. I'm not familiar with it, but there's a bunch of examples in nixpkgs already.

Reply:
https://nixos.org/manual/nixos/unstable/#sec-freeform-modules
hehe gotta read the documentation thoroughly... I'm working on it

Reply:
The way i have it now also adds `port` and `address` as environment variables, which is not really optimal, i think.
I can't `mkDefault` `services.rimgo.settings.PORT` to `cfg.settings.port`, because the freeform part of a submodule can't depend on the defined part without causing infinite recursion.

Also, i had this idea of implementing camelCase like for `services.vaultwarden.config`, but it's probably overkill and generally not really useful, because of the mental overhead of converting UPPER_SNAKE_CASE (from the rimgo docs) to camelCase only to have it converted back.

Reply:
i think the easiest way to do this would be to just call the defined options of the submodule `PORT` and `ADDRESS` directly.
Is it allowed to use non camelCase in options that directly translate to non camelCase options from somewhere else?

Reply:
I don't think there is any rule against it. And if there was, I'd probably be annoyed. The settings pattern is there so users can put in whatever values are necessary without having to modify the module. I don't see why the relaxed nature of it shouldn't extend to us *and* mean that non-camelCase is available since that's what rimgo calls for.